### PR TITLE
configure: Only use CXX when fuzzing

### PR DESCRIPTION
**Pull request by pdxjohnny**
Fixes: #1508

Signed-off-by: John Andersen <john.s.andersen@intel.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -242,7 +241,8 @@ AS_CASE(["x$with_fuzzing"],
         [AC_MSG_ERROR([Bad value for --with-fuzzing])])
 AM_CONDITIONAL([ENABLE_FUZZING],[test "x$with_fuzzing" != "xnone"])
 AS_IF([test "x$with_fuzzing" != "xnone"],
-      [AS_IF([test "x$enable_tcti_fuzzing" = xno],
+      [AC_PROG_CXX
```
Comment (tstruk) [inline configure.ac]: This throws warnings for me:
configure.ac:244: warning: AC_REQUIRE: `AC_PROG_CXX' was expanded before it was required

Comment (tstruk) [reply configure.ac]: and shouldn't there be a comma after AC_PROG_CXX?

Comment (pdxjohnny) [reply configure.ac]: > This throws warnings for me:
> configure.ac:244: warning: AC_REQUIRE: `AC_PROG_CXX' was expanded before it was required

I'm seeing this too, I just ran in CI I didn't run locally so I missed this. Sorry.

**Comment by casantos** [comment]
With this change I see the following warnings while running bootstrap:

configure.ac:288: warning: AC_REQUIRE: `AC_PROG_CXX' was expanded before it was required
configure.ac:288: http://www.gnu.org/software/autoconf/manual/autoconf.html#Expanded-Before-Required

**Pull request merged**